contracts/MechMarketplace.sol

```diff
@@ -381,132 +380,240 @@ contract MechMarketplace is IErrorsMarketplace {
             revert OutOfBounds(responseTimeout, minResponseTimeout, maxResponseTimeout);
         }
 
-        // Check for non-zero data
-        if (requestData.length == 0) {
+        uint256 numRequests = requestDatas.length;
+        // Check for zero value
+        if (numRequests == 0) {
             revert ZeroValue();
         }
 
         // Check priority mech
         address priorityMech = mapServiceIdMech[priorityMechServiceId];
         if (priorityMech == address(0)) {
             revert ZeroAddress();
-        }    
+        }
 
         // Check requester
         checkRequester(msg.sender, requesterServiceId);
 
-        // Calculate request Id
-        requestId = getRequestId(msg.sender, requestData, mapNonces[msg.sender]);
+        // Allocate set of requestIds
+        requestIds = new uint256[](numRequests);
 
-        // Update requester nonce
-        mapNonces[msg.sender]++;
+        // Get deliveryRate as priority mech max delivery rate
+        uint256 deliveryRate = IMech(priorityMech).maxDeliveryRate();
```

Comment:
Get mech delivery rate on the MM side at the beginning of the loop, as it's the same price that's used in each request.